### PR TITLE
[Feature] refactor seatunnel job run.

### PR DESCRIPTION
**Pull request by kalencaya**
## Purpose of this pull request
refactor seatunnel job run.

## Brief change log
- add `scaleph-engine-seatunnel` module for seatunnel job manage, eg. submit, schedule, cancel and so on.
- move `scaleph-api/cn.sliew.scaleph.api.schedule` package to `scaleph-core-scheduler` module.
- optimize seatunnel job run and schedule code for more readability, which not affect existing functions.
- cause some drawback, which reducing some schedule detail log. 
- split database schema to log and master.
- add local jdbc-to-jdbc examples.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If necessary, please update the documentation to describe the new feature.

### CODE REVIEW

File: scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelFlinkJob.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @SneakyThrows
+    @Override
+    protected void executeInternal(JobExecutionContext context) throws JobExecutionException {
+        JobDataMap dataMap = context.getJobDetail().getJobDataMap();
+        DiJobDTO job = (DiJobDTO) dataMap.get(Constants.JOB_PARAM_JOB_INFO);
+        DiJobRunVO jobRunParam = new DiJobRunVO();
+        jobRunParam.setJobId(job.getId());
+        jobRunParam.setClusterId(job.getClusterId());
+        seatunnelJobService.submit(jobRunParam);
+    }
```
Comment (gleiyu) [inline scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelFlinkJob.java:26]: resources is null in jobRunParam willl cause a java.lang.NullPointerException

Comment (gleiyu) [inline scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelFlinkJob.java:30]: resources is null in jobRunParam willl cause a java.lang.NullPointerException

File: scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelJobServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        job.setJobStepList(diJobStepService.listJobStep(jobId));
+        return job;
+    }
+
+    @Override
+    public void submit(DiJobRunVO jobRunParam) throws Exception {
+        // 1.执行任务和 flink 集群的绑定
+        diJobService.update(jobRunParam.toDto());
+        // 2.绑定任务和资源
+        diJobResourceFileService.bindResource(jobRunParam.getJobId(), jobRunParam.getResources());
```
[the remark this answers is not in the document]
Comment (gleiyu) [reply scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelJobServiceImpl.java:95]: 这段逻辑需要提到controller层

File: scaleph-api/src/main/java/cn/sliew/scaleph/api/controller/datadev/JobController.java [hunk 3]
```diff
@@ -642,119 +565,23 @@ public ResponseEntity<ResponseVO> publishJob(@PathVariable(value = "jobId") Long
     @ApiOperation(value = "运行任务", notes = "运行任务，提交至集群")
     @PreAuthorize("@svs.validate(T(cn.sliew.scaleph.common.constant.PrivilegeConstants).DATADEV_JOB_EDIT)")
     public ResponseEntity<ResponseVO> runJob(@RequestBody DiJobRunVO jobRunParam) throws Exception {
```
Comment (gleiyu) [inline scaleph-api/src/main/java/cn/sliew/scaleph/api/controller/datadev/JobController.java:567]: 缺少事务注解

File: scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelJobServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        // 3.获取任务信息
+        DiJobDTO diJobDTO = queryJobInfo(jobRunParam.getJobId());
+        if (JobTypeEnum.BATCH.getValue().equals(diJobDTO.getJobType().getValue())
+                && StringUtils.hasText(diJobDTO.getJobCrontab())) {
+            schedule(diJobDTO);
+            return;
+        }
```
Comment (gleiyu) [inline scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelJobServiceImpl.java]: job和cluster、resource的绑定只需要在前端任务启动那一刻绑定一次。
如果是周期任务且配置了crontab表达式，才会启动调度，否则只运行一次
这段逻辑需要提到controller层

**Review by gleiyu — CHANGES_REQUESTED** [review]

File: scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelFlinkJob.java (hunk 1 above)
Comment (kalencaya) [reply scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelFlinkJob.java:26]: already fixed

File: scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelJobServiceImpl.java (hunk 4 above)
Comment (kalencaya) [reply scaleph-engine/scaleph-engine-seatunnel/src/main/java/cn/sliew/scaleph/engine/seatunnel/service/impl/SeatunnelJobServiceImpl.java]: already fixed

**Pull request merged**